### PR TITLE
[image_picker] iOS: Make the system UI to be popover instead of fullscreen to prevent crashes on iOS 14.

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

See https://github.com/flutter/flutter/issues/67430, the crash can be prevented by this workaround: set the presentation style of the image picker to popover.

This PR also sets up some XCUITests to test the related scenarios.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/67430

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by teamdynamiteapp** [comment]
Hi, @cyanglaz.
I appreciating your commit.
I tried to your change on my app and was not crash on iOS14 👍 
But can not get image path from `imagePicker.getImage` result.
Do you recognize this issue?

**Comment by cyanglaz** [comment]
@teamdynamiteapp I didn't see this issue. Is it 100% reproducible or it is flaky?

File: packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m [hunk 1]
```diff
@@ -335,13 +335,19 @@ - (void)imagePickerController:(UIImagePickerController *)picker
...
+}
+
+- (void)handleImagePickerControllerDismissed {
   if (!self.result) {
     return;
   }
   self.result(nil);
   self.result = nil;
   _arguments = nil;
+  // Dismiss image picker after cleaning up can precent race a condition,
```
Comment (amirh) [inline packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m:348]: nit: typo (present) (also I'm not sure about the grammar, maybe s/can present race a condition/can lead to a race condition/)

File: packages/image_picker/image_picker/ios/UITests/ImagePickerExampleUITests.m [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <XCTest/XCTest.h>
+
+@interface ImagePickerExampleUITests : XCTestCase
+
+@end
+
+@implementation ImagePickerExampleUITests
```
Comment (amirh) [inline packages/image_picker/image_picker/ios/UITests/ImagePickerExampleUITests.m:11]: thanks!

**Review by amirh — APPROVED** [review]
Code looks good to me, LGTM once we figure out the issue @teamdynamiteapp's pointed out

**Comment by cyanglaz** [comment]
@teamdynamiteapp I found a potential race condition where the image picker controller can dismiss too fast that the result is not ready yet. I updated the code to dismiss the controller after all the results are ready. Could you try again to see if it fixes your issue? (I still can't repro the issue on my local tho)

I have updated the tests to run more times so it is more likely to catch race conditions.

File: packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m (hunk 1 above)
Comment (cyanglaz) [reply packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m:357]: done

**Comment by teamdynamiteapp** [comment]
@cyanglaz 

Thanks for your response.
I tried to your latest changes again.
Probably looks like everything ok for me 👍 

I looking forward to merge this PR.

**Comment by joniaiuser** [comment]
I tried this PR on iOS 14.0 flutter stable branch and it works perfectly.

However on iOS 13.6 using same iOS 11 Pro simulator it doesn't work. After selecting the image, nothing happens.

**Comment by joniaiuser** [comment]
e58a38d fixed the issue. Thanks

**Comment by azborovskyi** [comment]
It crashes on all iPads consistently with 
`'UIPopoverPresentationController (<UIPopoverPresentationController: 0x7ff884f30b30>) should have a non-nil sourceView or barButtonItem set before the presentation occurs.'`

**Comment by cyanglaz** [comment]
@azborovskyi Good catch. I didn't test it on iPad, will push a fix soon. 
@joniaiuser It was very unlikely that merging master fixed your issue as there wasn't any changes in the image picker. Maybe the issue you saw was some sort of race condition? Could you try more times to see if it happens?

**Comment by danielwalczak** [comment]
looking forward to this fix 🔥

**Comment by cyanglaz** [comment]
The crash has been fixed with the latest engine. I'm going to close this PR.

**Comment by skyberk** [comment]
Hi @cyanglaz I am grateful for your help.  The fixed engine means ios 14 update ? Can you share with us which update they fixed ?

**Comment by cyanglaz** [comment]
@skyberk No, I didn't mean an iOS update. I meant the the fix is in the flutter framework. You should see the fix in master channel.

**Comment by varun-horizon** [comment]
I am running latest flutter engine and its still crashing for me :/

**Comment by cyanglaz** [comment]
@varun-horizon Could you share more details? It was working for me. Could you share your `flutter doctor -v` and your `flutter run` command syntax? Maybe also the stacktrace?

**Comment by diegoveloper** [comment]
hey @cyanglaz thanks for fixing this. Do you know when it will be in `stable`?

**Comment by sanrixue** [comment]
I am running latest flutter engine and its still crashing for me too

**Pull request closed without merging**